### ISSUE TITLE
Tests run very slowly with app.asyncShutdown()

### ISSUE THREAD

**Issue by winsmith** [vapor/vapor#3231]
### Describe the issue

I followed a fix-it that asked me to convert all `app.shutdown()` calls in my tests to `try await app.asyncShutdown()` and now the tests take ages to run

### Vapor version

4.102.0

### Operating system and version

macOS 14.6.1

### Swift version

Swift Package Manager - Swift 5.10.0-dev

### Steps to reproduce

Here's an example of what my tests look like:

```swift
@testable import App
import XCTVapor

final class V3GroupsControllerTests: XCTestCase {
    var vaporApp: Application!

    override func setUp() async throws {
        vaporApp = try await Application.make(.testing)
        try configure(vaporApp)
        try await vaporApp.autoMigrate()

        // Create some test objects in the DB here...
    }

    override func tearDown() async throws {
        try await vaporApp.autoRevert()
        vaporApp.shutdown() // ⚠️ Instance method 'shutdown' is unavailable from asynchronous contexts; This can block the thread and should not be called in an async context; this is an error in the Swift 6 language mode
    }

    func testSomething() async throws {
        // ...
    }
}
```

Using Swift 6 compatibility mode, Xcode complains about `vaporApp.shutdown()` in my tests, asking me to instead use the `asyncShutdown` method. So I replaced the line with this:

```swift
try await vaporApp.asyncShutdown()
```

### Outcome

Converting my tests to asyncShutdown has made them factor 100 slower. Tests took around 60 seconds to run before, and now they take more than 3 hours, reproducably.

Also the console gets spammed with thousands of `shutdown` messages, one per line.

Am I holding it wrong? Any advice? 

### Additional notes

I tested this on a Mac with M1 Pro, an Intel Linux machine, and a GitHub Actions Ubuntu hosted runner.

**Comment by 0xTim** [comment]
@winsmith is this reproducible locally? If so do you know where the app is stuck? The only difference is that we call shutdown on any async lifecycle handlers, the rest should be exactly the same.

(Could you also try with the latest releases)

**Comment by winsmith** [comment]
Hi, thanks for the quick response!

- This is reproducible locally on both the command line and in Xcode. 
- I just updated to Vapor 4.103.2, the behaviour is still there
- The app/the test is not really stuck, it's just using a huge amount of CPU time and spamming the console. Eventually it *will* finish, just in a very unreasonable time.

Screenshot for your viewing pleasure:
<img width="1512" alt="Screenshot 2024-09-04 at 18 19 51" src="https://github.com/user-attachments/assets/31104127-4a27-476e-bc6f-673bad37d295">

**Comment by 0xTim** [comment]
That makes me think there's some infinite loop going on somewhere. Where is `shutdown` coming from? Are you doing anything with `LifecycleHandlers`?

**Comment by winsmith** [comment]
I'll check and get back to you. The string "shutdown" doesn't really appear in our code anywhere though...

Another thing that I just tried is disabling test parallelization, but that should already be disabled and setting it manually also didn't have an effect.

**Comment by 0xTim** [comment]
I've just checked Vapor's codebase and I can't find the string anywhere so it might be a dependency? That would be the best to work out what's going on

**Comment by winsmith** [comment]
No dependencies found yet but I had a hunch that it had to do with test parallelization, because when I run a single test, it works without issue. (This is not surprising because the tests expect complete reign over a local database)

Previously, all tests seem to have run in sequence, but switching the shutdown to async seems to have removed kind of intentional or unintentional semaphore. When I run the tests from the command line and force them to run in sequence with `swift test --parallel --num-workers 1`, the tests run perfectly.

I haven't yet found a way to force the tests to run in sequence in Xcode. Going to keep looking some more.

**Comment by 0xTim** [comment]
Hmm interesting, so they should work fine in parallel (we have a specific job in Vapor's CI to ensure the tests work in parallel). I'm assuming you're setting the port of each test `Application` to 0 to choose a random one?

**Comment by winsmith** [comment]
I think the tests *need* to run in sequence, because most of them access the database, and will destroy and rebuild them. Any ideas how to make the run in sequence in Xcode?

**Comment by 0xTim** [comment]
They should be serial by default unless you explicitly opt in

**Comment by 0xTim** [comment]
There's nothing in Vapor that logs out `shutdown`, especially as a print call because I'd expect to see other stuff in the log if it was a real log statement. So if you're happy with how things are, I'll close this out as there have been no other reports. Feel free to reopen if something crops up

**Comment by winsmith** [comment]
FYI, the issue is resolved with Xcode 16.0 RC
